File: Examples/Forms/ListBox/Sources/ListBox.cpp
#include <Pcf/Pcf>

using namespace System;
using namespace System::Windows::Forms;

namespace ListBoxExample {
  class Form1 : public Form {
  public:
    // The main entry point for the application.
    static void Main() {
      Application::EnableVisualStyles();
      Application::Run(Form1());
    }
    
    Form1() {
      this->Text = "ListBox example";
      this->StartPosition = FormStartPosition::Manual;
      this->Location = System::Drawing::Point(200, 200);
      this->ClientSize = System::Drawing::Size(360, 240);
      
      this->listBoxActionsLeft.Parent = *this;
      this->listBoxActionsLeft.Anchor = AnchorStyles::Top | AnchorStyles::Left | AnchorStyles::Bottom;
      this->listBoxActionsLeft.Bounds = System::Drawing::Rectangle(20, 20, 150, 200);
      this->listBoxActionsLeft.DoubleClick += pcf_delegate(const object& sender, const EventArgs& e) {
        if (this->listBoxActionsLeft.SelectedIndex != -1) {
          this->listBoxActionsRight.Items().Add(this->listBoxActionsLeft.SelectedItem());
          this->listBoxActionsRight.SelectedIndex = this->listBoxActionsRight.Items().Count-1;
          this->listBoxActionsLeft.Items().Remove(this->listBoxActionsLeft.SelectedItem());
        }
      };
      
      this->listBoxActionsRight.Parent = *this;
      this->listBoxActionsRight.Anchor = AnchorStyles::Top | AnchorStyles::Left | AnchorStyles::Bottom | AnchorStyles::Right;
      this->listBoxActionsRight.Bounds = System::Drawing::Rectangle(190, 20, 150, 200);
      this->listBoxActionsRight.DoubleClick += pcf_delegate(const object& sender, const EventArgs& e) {
        if (this->listBoxActionsRight.SelectedIndex != -1) {
          this->listBoxActionsLeft.Items().Add(this->listBoxActionsRight.SelectedItem());
          this->listBoxActionsLeft.SelectedIndex = this->listBoxActionsLeft.Items().Count-1;
          this->listBoxActionsRight.Items().Remove(this->listBoxActionsRight.SelectedItem());
        }
      };

      this->listBoxActionsLeft.Items().AddRange(Array<string> {"draw", "cut", "paste", "delete", "open", "close", "remove", "edit", "find", "increment", "decrement", "write", "read", "post", "build", "make", "release", "create", "choose", "erase"});
    }
    
  private:
    ListBox listBoxActionsLeft;
    ListBox listBoxActionsRight;
  };
}

pcf_startup (ListBoxExample::Form1)

File: CSharp/Forms/Examples/ListBox/ListBox.cs
using System;
using System.Windows.Forms;

namespace ListBoxExample {
  class Form1 : Form {
    public Form1() {
      this.StartPosition = FormStartPosition.Manual;
      this.Text = "ListBox example";
      this.Location = new System.Drawing.Point(200, 200);
      this.ClientSize = new System.Drawing.Size(360, 240);

      this.listBox1.Parent = this;
      this.listBox1.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Bottom;
      this.listBox1.Bounds = new System.Drawing.Rectangle(20, 20, 150, 200);
      this.listBox1.DoubleClick += delegate(object sender, EventArgs e) {
        if (this.listBox1.SelectedIndex != -1) {
          this.listBox2.Items.Add(this.listBox1.SelectedItem);
          this.listBox2.SelectedIndex = this.listBox2.Items.Count-1;
          this.listBox1.Items.Remove(this.listBox1.SelectedItem);
        }
      };

      this.listBox2.Parent = this;
      this.listBox2.Anchor = AnchorStyles.Top | AnchorStyles.Left | AnchorStyles.Bottom | AnchorStyles.Right;
      this.listBox2.Bounds = new System.Drawing.Rectangle(190, 20, 150, 200);
      this.listBox2.DoubleClick += delegate(object sender, EventArgs e) {
        if (this.listBox2.SelectedIndex != -1) {
          this.listBox1.Items.Add(this.listBox2.SelectedItem);
          this.listBox1.SelectedIndex = this.listBox1.Items.Count-1;
          this.listBox2.Items.Remove(this.listBox2.SelectedItem);
        }
      };

      this.listBox1.Items.AddRange(new string[] {"draw", "cut", "paste", "delete", "open", "close", "remove", "edit", "find", "increment", "decrement", "write", "read", "post", "build", "make", "release", "create", "choose", "erase"});
    }

    private ListBox listBox1 = new ListBox();
    private ListBox listBox2 = new ListBox();
  }

  class MainClass {
    [STAThread]
    public static void Main() {
      Application.EnableVisualStyles();
      Application.Run(new Form1());
    }
  }
}
